### DISCUSSION TITLE
blog/graphql-cursor-pagination-with-postgresql

### DISCUSSION THREAD

**Discussion by giscus** [the-guild-org/the-guild-website#1100] [Docs Discussions]
# blog/graphql-cursor-pagination-with-postgresql

An SQL based approach for understanding and implementing GraphQL Cursor Pagination.

https://the-guild.dev/blog/graphql-cursor-pagination-with-postgresql

**Comment by bondowe** [comment]
Hi,
shouldn't the second predicate for created_at be greaterthan (>) rather than (>=) ?

e.g.:

WHERE
  (
    "created_at" = '2023-01-11T10:32:07.853915+00:00'
    AND "id" > '628995bf-2907-49d1-a36f-978566a053c4'
  )
  OR "created_at" > '2023-01-11T10:32:07.853915+00:00'

**Reply by bhasden** [reply]
Ran into the same thing. I believe you're correct, the >= operation will not page properly.

EDIT: I submitted a PR to fix the example in the blog post.

**Reply by bhasden** [reply]
Well, that was quick. [The PR](https://github.com/the-guild-org/website/pull/1253) has been merged, however, the blog post doesn't appear to be updated yet.

**Comment by jedwards1211** [comment]
You can use tuple comparison to be more succinct:

`(created_at, id) > ($1, $2)`

**Reply by bondowe** [reply]
No. The first created_at is rightly checked for equality.
It's the second one that should use >.

**Reply by jedwards1211** [reply]
`(a, b) > (c, d)` is equivalent to `a > c OR (a = c AND b > d)`...it's like how string comparison is comparing one letter at a time under the hood, tuple comparison compares one element at a time

**Reply by bondowe** [reply]
Thank you. I wasn't aware of that.

**Comment by ziplizard** [comment]
How to get count without a entirely new query?
